services/hanke-service/src/main/resources/db/changelog/changesets/086-change-car-lane-nuisance-enum.sql

```diff
@@ -14,41 +14,40 @@ UPDATE hankealue
                 jsonb_set(
                     applicationData,
                     '{areas}',
-                    (
-                        SELECT jsonb_agg(
-                            jsonb_set(
-                                area,
-                                '{kaistahaitta}',
-                                (
+                    jsonb_agg(
+                        CASE
+                            WHEN (area #>> '{kaistahaitta}') IS NOT NULL THEN
+                                jsonb_set(
+                                    area,
+                                    '{kaistahaitta}',
                                     CASE area->>'kaistahaitta'
-                                        WHEN 'VAHENTAA_KAISTAN_YHDELLA_AJOSUUNNALLA' THEN
-                                            '"YKSI_KAISTA_VAHENEE"'::jsonb
-                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_KAISTAN_KAHDELLA_AJOSUUNNALLA' THEN
-                                            '"YKSI_KAISTA_VAHENEE_KAHDELLA_AJOSUUNNALLA"'::jsonb
-                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_KAHDELLA_AJOSUUNNALLA' THEN
-                                            '"USEITA_KAISTOJA_VAHENEE_AJOSUUNNILLA"'::jsonb
-                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_LIITTYMIEN_ERI_SUUNNILLA' THEN
-                                            '"USEITA_AJOSUUNTIA_POISTUU_KAYTOSTA"'::jsonb
+                                        WHEN 'VAHENTAA_KAISTAN_YHDELLA_AJOSUUNNALLA' THEN '"YKSI_KAISTA_VAHENEE"'::jsonb
+                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_KAISTAN_KAHDELLA_AJOSUUNNALLA' THEN '"YKSI_KAISTA_VAHENEE_KAHDELLA_AJOSUUNNALLA"'::jsonb
+                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_KAHDELLA_AJOSUUNNALLA' THEN '"USEITA_KAISTOJA_VAHENEE_AJOSUUNNILLA"'::jsonb
+                                        WHEN 'VAHENTAA_SAMANAIKAISESTI_USEITA_KAISTOJA_LIITTYMIEN_ERI_SUUNNILLA' THEN '"USEITA_AJOSUUNTIA_POISTUU_KAYTOSTA"'::jsonb
                                         ELSE area->'kaistahaitta'
                                     END
                                 )
-                            )
-                        )
-                        FROM jsonb_array_elements(applicationData->'areas') area
+                            ELSE area
+                        END
                     )
                 )
             ELSE applicationData
-        END as updated_data
-    FROM applications
+        END AS updated_data
+    FROM applications,
+         jsonb_array_elements(applicationData->'areas') AS area
+    WHERE applicationData IS NOT NULL
+    GROUP BY id, applicationData
 )
+
 UPDATE applications
-SET applicationData = areas_array.updated_data
-FROM areas_array
-WHERE applications.id = areas_array.id;
+SET applicationData = updated_areas.updated_data
+    FROM updated_areas
+WHERE applications.id = updated_areas.id;
```

Comment:
I would be kind of happy to limit this migration to WHERE `applications.applicationtype = 'EXCAVATION_NOTIFICATION`. We don't have to worry about the cable reports or any of their quirks. It would probably make the query simpler as well, since we could make more assumptions about the data structure.